### PR TITLE
Move plugin implementation into java source set

### PR DESCRIPTION
**Pull request by ferozco**
## After this PR

==COMMIT_MSG==
Move plugin implementation into java source set
==COMMIT_MSG==

### CODE REVIEW

**Comment by iamdanfox** [comment]
Lol I just started this too... complication:

```
$ find . -path '**main/*.groovy'
./gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/pod/PodVolumeDefinition.groovy
./gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/pod/PodServiceDefinition.groovy
./gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/ProductId.groovy
./gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/tasks/CopyLauncherBinariesTask.groovy
./gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/util/EmitFiles.groovy
```

**Comment by ferozco** [comment]
ya I'm fixing up the few classes that are still in groovy

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java [hunk 1]
```diff
@@ -109,28 +122,31 @@ public void apply(Project project) {
...
-                                .map(File::getName)
-                                .collect(Collectors.joining(" "));
-                        task.getManifest()
-                                .getAttributes()
-                                .put(
-                                        "Class-Path",
-                                        classPath
-                                                + " "
-                                                + task.getArchiveFileName().get());
+                    task.doFirst(new Action<Task>() {
```
Comment (iamdanfox) [inline gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java:125]: boom, error-prone catches another one!

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java [hunk 2]
```diff
@@ -253,7 +254,10 @@ final void createManifest() throws Exception {
                 return mergedDependency;
             });
         });
-        List<ProductDependency> productDeps = new ArrayList<>(allProductDependencies.values());
+        List<ProductDependency> productDeps = allProductDependencies.values().stream()
+                .sorted(Comparator.comparing(ProductDependency::getProductGroup)
+                        .thenComparing(ProductDependency::getProductName))
```
Comment (iamdanfox) [inline gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java:259]: nice

**Review by iamdanfox — APPROVED** [review]
Oh my dayys so happy to have real working static analysis for this stuff

**Pull request merged**